src/video.rs

```diff
@@ -302,6 +312,15 @@ impl Video {
         let worker = std::thread::spawn(move || {
             let mut clear_subtitles_at = None;
 
+            macro_rules! cleanup {
```

Comment:
The cleanup shouldn't be needed inside the worker thread - if errors occur here they are recovered gracefully (without explicitly stopping the pipeline).
Moreover, once `spawn` is reached, no subsequent early returns are possible (since it's at the end) - so from hereon the construct *will* return and thus proper pipeline cleanup is handled by the `Drop` impl.

Reply:
Gotcha, that makes sense. I reverted those parts \:)